### DISCUSSION TITLE
tflite & scorer & audio files

### DISCUSSION THREAD

**Discussion by i2rllc** [coqui-ai/STT#2335] [Q&A]
This is extremely novice but what directory do the tflite, scorer and audio files get stored to to execute STT? I put it in coqui-stt-venv and coqui-stt-venv/bin and it's not executing so clearly the files are in the wrong directory; please help. TYIA

**Comment by HarikalarKutusu** [comment]
What are you trying to do? What is your environment?

In any case, you can put your data anywhere and point to them with command-line arguments in training for example.

You can find the related arguments by using --help with each command and they are explained here:
https://stt.readthedocs.io/en/latest/TRAINING_FLAGS.html

For example, in the notebook, which ran on Google Colab, I kept the data on Google Drive and expand it to the Colab instance (Ubuntu) somewhere and pointed to them during calling the related commands:
https://github.com/HarikalarKutusu/common-voice-tr-experiments/blob/main/colab-notebooks/tr/v8.0/r4/stt-train-cv-tr-v8.0-r4-s5.ipynb

Or for inference, for the Voice Chess application, I used a server and put it data somewhere, and point to them:
https://github.com/HarikalarKutusu/3d-voice-chess/blob/897ab6598460b5a5cae10f9f69fc1f10c6426efa/voice-chess-server/server.js#L29

**Reply by i2rllc** [reply]
@HarikalarKutusu Right now, I'm just trying to get STT to run locally. Looks like I do not have the tfile, scorer files in the right place. I'm new to linux and programming generally so that's where I'm looking for help right now. Thank you
![image](https://user-images.githubusercontent.com/122296028/213160701-ea44fdcd-6024-4c24-a129-2665486d867b.png)

**Reply by HarikalarKutusu** [reply]
I see. You are at root and you are trying to read from there. Perhaps it would be better to work like this:
```
~/
   stt/
   audio/
      *.wav
   models/
     en/
        model.tflite
        huge-vocabulary.scorer
     xx/
```
Then call it like this:
```sh
stt --model ~/models/en/model.tflite --scorer ~/models/en/huge-vocabulary.scorer --audio ~/audio/test.wav
```
And AFAIK you can only use .wav files as input.

**Reply by i2rllc** [reply]
Thanks, absolutely path worked but now I have another error. Any idea what's happening here? For reference, this is the model I'm using.
https://coqui.ai/english/coqui/v1.0.0-huge-vocab#download
![image](https://user-images.githubusercontent.com/122296028/213281512-c1838d51-8086-41c5-a490-44a809b4ed4d.png)

**Reply by wasertech** [reply]
1. Don't share logs using screenshots.
You can use [gist](https://gist.github.com) or something like dpaste to share your logs.
2. The error message is pretty explicit.
You are missing `sox`. Install it using your package manager.
On Ubunutu use something like `apt-get install libasound2-plugins libasound2-python libsox-fmt-all libsox-dev sox`
3. Sox is needed here because your wav files are not in the correct format.
Wav files must be using 16K Hz sample rate. If you install sox, it will convert them for you but it's better to make sure your audio are in the correct format manually.
Using `soxi` on your file, it should look something like that:
```bash
❯ soxi audio.wav

Input File     : 'audio.wav'
Channels       : 1
Sample Rate    : 16000
Precision      : 16-bit
Duration       : 00:00:07.68 = 122880 samples ~ 576 CDDA sectors
File Size      : 246k
Bit Rate       : 256k
Sample Encoding: 16-bit Signed Integer PCM
```
Read the [docs about audio format](https://stt.readthedocs.io/en/latest/TRAINING_INTRO.html?highlight=audio%20format#data-format).

**Reply by i2rllc** [reply]
@wasertech Thank you! 

1) Why use gist or dpaste? What does it do that screenshot doesn't?
2) This is probably going to be very stupid question but how do I send the output to a file. I tried executing the command followed by **> filename.txt** but it didn't actually save anything? I must be missing something. Thank you again

**Reply by wasertech** [reply]
> Why use gist or dpaste? What does it do that screenshot doesn't?

You want our help? You share your logs the proper way. It's that simple. We need to be able to copy and paste your logs, search trough them, etc.

> This is probably going to be very stupid question but how do I send the output to a file. I tried executing the command followed by > filename.txt but it didn't actually save anything

Try like this:
```bash
❯ stt --model output_graph.tflite --scorer kenlm.scorer  --audio audio.wav > transcript.txt
Loading model from file output_graph.tflite
TensorFlow: v2.8.0-8-g06c8fea58fd
 Coqui STT: v1.3.0-0-g148fa743
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
Loaded model in 0.00897s.
Loading scorer from files kenlm.scorer
Loaded scorer in 0.0037s.
Running inference.
Inference took 1.075s for 3.580s audio file.
❯ cat transcript.txt
───────┬───────────────────────────────────────────────────────────────────────────
       │ File: transcript.txt
───────┼───────────────────────────────────────────────────────────────────────────
   1   │ bonjour
───────┴───────────────────────────────────────────────────────────────────────────
```

You can also checkout [`listen`](https://gitlab.com/waser-technologies/technologies/listen) which allows to directly transcribe into a file like so:

```bash
❯ listen -f audio.wav
Filename                       Duration(s)         
audio.wav                     3.580               

❯ cat audio.txt
Bonjour.
```

[Read more](https://github.com/coqui-ai/STT/discussions/2337)

**Reply by i2rllc** [reply]
@wasertech Thank you. I am new to this so your effort to teach me is appreciated.

**Reply by HarikalarKutusu** [reply]
> Why use gist or dpaste? What does it do that screenshot doesn't?

For example, I saw your post yesterday night, but I was on mobile. I could not see the details in the image, so I could not answer.

**Reply by i2rllc** [reply]
@HarikalarKutusu understood, thank you
